apiserver/apiserver/apiserver.py

```diff
@@ -19,6 +20,26 @@
 routes = web.RouteTableDef()
 
 
+with open(os.environ.get('HAIL_JWT_SECRET_KEY_FILE') or '/jwt-secret/secret-key') as f:
+    jwtclient = hj.JWTClient(f.read())
+
+
+def authenticated_users_only(fun):
+    def wrapped(request, *args, **kwargs):
+        encoded_token = request.cookies.get('user')
+        if encoded_token is not None:
+            try:
+                userdata = jwtclient.decode(encoded_token)
+                if 'userdata' in fun.__code__.co_varnames:
+                    return fun(request, *args, userdata=userdata, **kwargs)
+                return fun(request, *args, **kwargs)
+            except jwt.exceptions.DecodeError:
+                pass
+        raise web.HTTPUnauthorized(headers={'WWW-Authenticate': 'Bearer'})
+    wrapped.__name__ = fun.__name__
```

Comment:
Out of curiosity why not use functools @wraps for this. Just lighter?

Reply:
ignorance. looks like that also copies the docstring, I'll switch